main.go

```diff
@@ -61,18 +64,30 @@ func main() {
 		log.Panicln("Error creating athenz domain client:", err.Error())
 	}
 
-	c := controller.NewController(*dnsSuffix, istioClient, k8sClient, adClient)
+	adResyncInterval, err := time.ParseDuration(*adResyncIntervalRaw)
+	if err != nil {
+		log.Panicln("Error parsing ad-resync-interval duration:", err.Error())
+	}
+
+	crcResyncInterval, err := time.ParseDuration(*crcResyncIntervalRaw)
+	if err != nil {
+		log.Panicln("Error parsing crc-resync-interval duration:", err.Error())
+	}
+
+	c := controller.NewController(*dnsSuffix, istioClient, k8sClient, adClient, adResyncInterval, crcResyncInterval)
 
-	stopChan := make(chan struct{})
-	go c.Run(stopChan)
+	stopCh := make(chan struct{})
+	go c.Run(stopCh)
 
-	signalChan := make(chan os.Signal, 1)
-	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
 	for {
 		select {
-		case <-signalChan:
-			log.Println("Shutdown signal received, exiting...")
-			close(stopChan)
+		case <-signalCh:
+			log.Println("Shutdown signal received, stopping controllers...")
+			close(stopCh)
+			<-signalCh
```

Comment:
should we simply do a time.Sleep(1s)  instead of waiting for one more SIG?

Reply:
@prabushyam Sure, I have updated it. The sleep is another convention I noticed people doing here as well.

Reply:
I am not a fan of time.Sleep either. But, the process should get terminated on SIGTERM, and not wait for SIGKILL IMHO.